cpp/tests/strings/replace_tests.cpp

```diff
@@ -352,6 +345,56 @@ TEST_F(StringsReplaceTest, ReplaceMulti)
   }
 }
 
+TEST_F(StringsReplaceTest, ReplaceMultiLong)
+{
+  auto input = cudf::test::strings_column_wrapper(
+    {"This string needs to be very long to trigger the long-replace internal functions.",
+     "01234567890123456789012345678901234567890123456789012345678901234567890123456789",
+     "01234567890123456789012345678901234567890123456789012345678901234567890123456789",
+     "Test string for overlap check: bananaápple bananá ápplebananá banápple ápple bananá",
+     "",
+     ""},
+    {1, 1, 1, 1, 0, 1});
```

Comment:
should this calculate the size and assert it is > `AVG_CHAR_BYTES_THRESHOLD` to prevent losing coverage if `AVG_CHAR_BYTES_THRESHOLD` is ever changed?

Reply:
I think that is a good idea.

Reply:
Not that important. At most I would comment near the value that this test attempts to build a string large enough to exercise it. I wouldn't hold up this PR over it.